src/racer.c

```diff
@@ -2273,7 +2519,154 @@ void func_80054110(Object *obj, Object_Racer *racer, s32 updateRate, f32 updateR
     }
 }
 
-GLOBAL_ASM("asm/non_matchings/racer/func_8005492C.s")
+void func_8005492C(Object* obj, Object_Racer* racer, s32 updateRate, f32 updateRateF) {
+    s32 xStick;
+    s32 stickMultiplier;
+    s32 surfaceType;
+    s32 sp38;
+    f32 temp_f0_2;
+    f32 velSquare;
+    f32 pad;
+    f32 traction;
+    f32 forwardVel;
+    f32 weight;
+    f32 vel;
+    f32 multiplier;
+    s32 rawStick;
+    s32 velocityS;
+    s32 temp;
+    u16 temp2;
+
+    velSquare = racer->velocity * racer->velocity;
+    racer->unk84 = 0.0f;
+    racer->unk88 = 0.0f;
+    if (racer->velocity < 0.0f) {
+        velSquare = -velSquare;
+    }
+    sp38 = 0;
+    multiplier = handle_racer_top_speed(obj, racer);
+    if (racer->y_rotation_vel > 0x1C00 || racer->y_rotation_vel < -0x1C00) {
+        sp38 = 1;
+    }
+    rawStick = gCurrentStickX;
+    stickMultiplier = 6;
+    if (gCurrentRacerInput & B_BUTTON) {
+        stickMultiplier = 12;
+    }
+    if (gCurrentRacerInput & B_BUTTON && racer->drift_direction != 0) {
+        stickMultiplier = 18;
+    }
+    xStick = 0;
+    if (racer->velocity < -0.3) {
+        xStick = rawStick * stickMultiplier;
+    }
+    if (racer->velocity > 0.3) {
+        xStick = -rawStick * stickMultiplier;
+    }
+    xStick *= updateRate;
+    temp = (s32) ((xStick >> 1) * gCurrentRacerHandlingStat); 
+    temp2 = temp;
+    racer->unk1A0 -= temp2;
+    handle_car_steering(racer);
+    racer->lateral_velocity *= 0.9;
+    surfaceType = SURFACE_DEFAULT;
+    traction = 0.0f;
+    if (racer->wheel_surfaces[0] != SURFACE_NONE) {
+        traction += gSurfaceTractionTable[racer->wheel_surfaces[0]];
+        if (racer->wheel_surfaces[0] > SURFACE_DEFAULT) {
+            surfaceType = racer->wheel_surfaces[0];
+        }
+    }
+    if (racer->buoyancy != 0.0) {
+        traction = 0.02f;
+    }
+    gCurrentSurfaceType = surfaceType;
+    racer->drift_direction = 0;
+    racer->unk10C = 0;
+    if (surfaceType == 4) {
+        multiplier = 0.25f;
+        racer->lateral_velocity = 0.0f;
+    }
+    if (racer->boostTimer == 0 && surfaceType == 3) {
```

Comment:
Haven't checked in a debugger, but does this imply that `SURFACE_UNK3` is for driving on zippers?

Reply:
I assume so, but having tested ingame, I cannot verify for sure, so I'll keep it as UNK3 for now.